tools/toolchains/mbed_toolchain.py

```diff
@@ -397,14 +397,27 @@ def get_arch_file(self, objects):
 
     # THIS METHOD IS BEING CALLED BY THE MBED ONLINE BUILD SYSTEM
     # ANY CHANGE OF PARAMETERS OR RETURN VALUES WILL BREAK COMPATIBILITY
-    def compile_sources(self, resources, inc_dirs=None):
+    def compile_sources(self, resources, inc_dirs=None, exclude_paths=None):
         # Web IDE progress bar for project build
         files_to_compile = (
             resources.get_file_refs(FileType.ASM_SRC) +
             resources.get_file_refs(FileType.C_SRC) +
             resources.get_file_refs(FileType.CPP_SRC)
         )
-        self.to_be_compiled = len(files_to_compile)
+        # Remove files from paths to be excluded from the build and create
+        # a compilation queue.
+        compile_queue = (
+            files_to_compile
+            if not exclude_paths
+            else [
+                file_to_compile
+                for exclude_path in exclude_paths
```

Comment:
I'm not sure this list comprehension does what you want. It works ok when there is a single exclude path, but when there are multiple exclude paths the file to compile with be included once for every exclude path.

I don't think it would be a bad idea to break out this functionality into a small function and unit test.

Reply:
The scope of this change is to provide a short-term workaround to maintain Mbed OS 2 compatibility until legacy build support is removed. 
If the current code works when there is a single exclude path then let's add a comment to that effect and let's convert `exclude_paths` to a single `exclude_path`.